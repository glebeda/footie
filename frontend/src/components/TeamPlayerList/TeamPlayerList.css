.team-player-list-container {
  margin-top: 0.6rem;
  overflow-y: auto;
  scroll-behavior: smooth;
  padding: 1rem;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  border-radius: 8px;
  width: 100%;
  box-sizing: border-box; 
}

.team-player-list-container.lights {
  border-top: 5px solid #fbc02d;
}

.team-player-list-container.darks {
  border-top: 5px solid #607d8b;
}

.team-title {
  text-align: center;
  font-weight: bold;
  padding: 0.5rem 0;
  background-color: #f5f5f5;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
}

.team-title .team-icon {
  margin-right: 8px;
}

.team-row {
  &:nth-of-type(odd) {
    background-color: #f9f9f9;
  }
}

.team-title > h6 {
  margin: 0;
}
