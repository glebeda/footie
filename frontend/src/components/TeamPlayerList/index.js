import React from 'react';
import './TeamPlayerList.css';
import WbSunnyIcon from '@mui/icons-material/WbSunny';
import Brightness2Icon from '@mui/icons-material/Brightness2';
import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Paper, Typography } from '@mui/material';

function TeamPlayerList({ teamName, players }) {
  return (
    <TableContainer className={`team-player-list-container ${teamName.toLowerCase()}`} component={Paper}>
      <div className="team-title">
        {teamName === 'Lights' ? (
          <WbSunnyIcon className="team-icon" />
        ) : (
          <Brightness2Icon className="team-icon" />
        )}
        <Typography variant="h6" component="div">
          {teamName}
        </Typography>
      </div>
      <Table aria-label={`${teamName} table`}>
        <TableHead>
          <TableRow>
            <TableCell className='fixed-width'>#</TableCell>
            <TableCell>Name</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {players.map((player, index) => (
            <TableRow key={`player-${player.playerId}`} className="team-row">
              <TableCell className='fixed-width' component='th' scope='row'>{index + 1}</TableCell>
              <TableCell>{player.name}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default TeamPlayerList;
