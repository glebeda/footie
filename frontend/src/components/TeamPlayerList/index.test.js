import React from 'react';
import { render, screen } from '@testing-library/react';
import TeamPlayerList from './index';

describe('TeamPlayerList', () => {
  const players = [
    { playerId: '1', name: 'Chuchundra' },
    { playerId: '2', name: 'Buboo' },
    { playerId: '3', name: 'Kookoosya' },
  ];

  it('renders the team title', () => {
    render(<TeamPlayerList teamName="Lights" players={players} />);
    expect(screen.getByText('Lights')).toBeInTheDocument();
  });

  it('renders the correct number of player rows', () => {
    render(<TeamPlayerList teamName="Lights" players={players} />);
    const playerRows = screen.getAllByRole('row');
    expect(playerRows).toHaveLength(players.length + 1);
  });

  it('renders the player names correctly', () => {
    render(<TeamPlayerList teamName="Lights" players={players} />);
    players.forEach(player => {
      expect(screen.getByText(player.name)).toBeInTheDocument();
    });
  });

  it('matches the snapshot with players', () => {
    const { asFragment } = render(<TeamPlayerList teamName="Lights" players={players} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot without players', () => {
    const { asFragment } = render(<TeamPlayerList teamName="Lights" players={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
