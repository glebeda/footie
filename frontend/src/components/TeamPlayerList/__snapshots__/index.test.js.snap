// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPlayerList matches the snapshot with players 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root team-player-list-container lights css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <div
      class="team-title"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium team-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="WbSunnyIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79zM4 10.5H1v2h3zm9-9.95h-2V3.5h2zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79zM20 10.5v2h3v-2zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6m-1 16.95h2V19.5h-2zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41z"
        />
      </svg>
      <div
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Lights
      </div>
    </div>
    <table
      aria-label="Lights table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium fixed-width css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            #
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root team-row css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium fixed-width css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            1
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Chuchundra
          </td>
        </tr>
        <tr
          class="MuiTableRow-root team-row css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium fixed-width css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Buboo
          </td>
        </tr>
        <tr
          class="MuiTableRow-root team-row css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium fixed-width css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            3
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Kookoosya
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`TeamPlayerList matches the snapshot without players 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root team-player-list-container lights css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <div
      class="team-title"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium team-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="WbSunnyIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79zM4 10.5H1v2h3zm9-9.95h-2V3.5h2zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79zM20 10.5v2h3v-2zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6m-1 16.95h2V19.5h-2zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41z"
        />
      </svg>
      <div
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Lights
      </div>
    </div>
    <table
      aria-label="Lights table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium fixed-width css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            #
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
